### PR TITLE
feat: optional telemetry

### LINKED ISSUE #69: Add optional telemetry
- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

## Description

Add optional telemetry deployments for newly created subscriptions.

### Additional context

### PR DESCRIPTION
**Pull request by matt-FFFFFF**
# Overview/Summary

Adds optional telemetry

## This PR fixes/adds/changes/removes

1. fixes #69
2. Modifies docs remove specific module version

## Testing Evidence

* Added bitfield value check to integration tests.
* Added integration test for disable telemetry
* Deployment test run for integration test only: https://github.com/Azure/terraform-azurerm-lz-vending/actions/runs/2711975650

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/terraform-azurerm-lz-vending/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/terraform-azurerm-lz-vending/issues), for tracking and closure.
- [x] Run `make docs` and `make fmt` to update the documentation and format your code.
- [x] Created unit and deployment tests and provided evidence.
- [x] Updated relevant and associated documentation.
- [ ] (Optionally) Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located in the [Enterprise-Scale repo](https://github.com/Azure/Enterprise-Scale) in the directory: `/docs/wiki/whats-new.md`)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,9 +28,9 @@ Please see the content in the [wiki](https://github.com/Azure/terraform-azurerm-
 ## Example
 
 ```terraform
-module "alz_landing_zone" {
+module "lz_vending" {
   source  = "Azure/lz-vending/azurerm"
-  version = "~>0.1.0"
+  version = "<version>" # change this to your desired version
```
Comment (krowlandson) [inline README.md:33]: I get why you're doing this from a code maintenance perspective, but does this make it harder for consumers? I don't find this a big overhead to update on each release.

**Review by krowlandson — APPROVED** [review]
LGTM, although cannot really comment on all the Go tests 😆

File: README.md (hunk 1 above)
Comment (matt-FFFFFF) [reply README.md]: Added a link to the hashicorp docs page, but left as placeholder

File: README.md [hunk 2]
```diff
@@ -118,6 +122,31 @@ No required inputs.
...
+Description: To disable tracking, we have included this variable with a simple boolean flag.  
+The default value is `false` which does not disable the telemetry.  
+If you would like to disable this tracking, then simply set this value to true and this module will not create the telemetry tracking resources and therefore telemetry tracking will be disabled.
+
+For more information, see the [wiki](https://aka.ms/lz-vending/tf/telemetry)
+
+E.g.
+
+```terraform
+module "lz-vending" {
```
[the remark this answers is not in the document]
Comment (matt-FFFFFF) [reply README.md]: changed

**Review by krowlandson — APPROVED** [review]
LGTM

**Pull request merged**